pkg/cmds/restore_volumesnapshot.go

```diff
@@ -0,0 +1,191 @@
+		pvcData = []PVC{}
+		for _, vol := range restoreSession.Spec.Target.VolumeClaimTemplates {
+			pvcData = append(pvcData, PVC{podOrdinal: types.Int32P(0), pvcName: vol.Name, pvc: &vol})
+
+		}
+	} else {
+		pvcData = []PVC{}
+		for i := int32(0); i < types.Int32(restoreSession.Spec.Target.Replicas); i++ {
+			for _, vol := range restoreSession.Spec.Target.VolumeClaimTemplates {
+				pvcData = append(pvcData, PVC{pvcName: vol.Name, podOrdinal: types.Int32P(i), pvc: &vol})
+			}
+		}
+
+	}
+
+	objectMeta := []metav1.ObjectMeta{}
+
+	for _, data := range pvcData {
+		pvc := opt.getPVCDefinition(data)
+		_, err := opt.kubeClient.CoreV1().PersistentVolumeClaims(opt.namespace).Get(data.pvc.Name, metav1.GetOptions{})
+		if err != nil {
+			pvc, err := opt.kubeClient.CoreV1().PersistentVolumeClaims(opt.namespace).Create(pvc)
+			if err != nil {
+				return err
+			}
+			objectMeta = append(objectMeta, pvc.ObjectMeta)
+
+		} else {
+			// write failure event
+			return opt.setRestoreSessionFailed(restoreSession, fmt.Sprintf("PVC %s is already exists", data.pvc.Name))
+		}
+	}
+
+	for i, pvc := range pvcData {
+		storageClass, err := opt.kubeClient.StorageV1().StorageClasses().Get(types.String(pvc.pvc.Spec.StorageClassName), metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if *storageClass.VolumeBindingMode != storage_api_v1.VolumeBindingImmediate {
+			// write failure event
+			return opt.setRestoreSessionFailed(restoreSession, fmt.Sprintf("VolumeBindingMode is equal to %s because of phase is Pending", storage_api_v1.VolumeBindingWaitForFirstConsumer))
+		}
+		err = util.WaitUntilPVCReady(opt.kubeClient, objectMeta[i])
+		if err != nil {
+			return err
+		}
+		// Update Backup Session
+		o := status.UpdateStatusOptions{
+			KubeClient:     opt.kubeClient,
+			StashClient:    opt.stashClient.(*cs.Clientset),
+			Namespace:      opt.namespace,
+			RestoreSession: opt.name,
+		}
+		restoreOutput := restic.RestoreOutput{
+			HostRestoreStats: v1beta1.HostRestoreStats{
+				Hostname: pvc.pvcName,
+				Phase:    v1beta1.HostRestoreSucceeded,
+			},
+		}
+		// Volume Snapshot complete. Read current time and calculate total backup duration.
+		endTime := time.Now()
+		restoreOutput.HostRestoreStats.Duration = endTime.Sub(startTime).String()
+		err = o.UpdatePostRestoreStatus(&restoreOutput)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (opt *VSoption) getPVCDefinition(data PVC) *corev1.PersistentVolumeClaim {
+	data.pvc.Name = fmt.Sprintf("%v-%v", data.pvcName, types.Int32(data.podOrdinal))
+	inputs := make(map[string]string, 0)
+	inputs["PVC_NAME"] = data.pvcName
+	fmt.Println(strconv.Itoa(int(types.Int32(data.podOrdinal))))
+	inputs["POD_ORDINAL"] = strconv.Itoa(int(types.Int32(data.podOrdinal)))
+	err := resolve.ResolvePVCSpec(data.pvc, inputs)
+	if err != nil {
+		return nil
+	}
+	return data.pvc
+}
+
+func (opt *VSoption) setRestoreSessionFailed(restoreSession *v1beta1.RestoreSession, message string) error {
```

Comment:
Don't need this function. Don't fail `RestoreSession` from here.  Just update status for respective host.